src/lib/Parser.hs

```diff
@@ -260,6 +264,52 @@ topLet = do
         k = if v == mkName "eff" then EffectRowKind else TypeKind
         uTyKind = WithSrc (Just pos) $ UPrimExpr $ TCExpr k
 
+interfaceDef :: Parser [UDecl]
+interfaceDef = do
+  keyWord InterfaceKW
+  -- TODO: allow multi-parameter type classes
+  interfaceName <- upperName <|> symName
+  (oneAnnBinder, pos) <- withPos annBinder
+  keyWord WhereKW
+  record <- withSrc $ URecordTy <$> interfaceRecordFields ":"
+  let consName = mkInterfaceConsName interfaceName
+  let (varTypeName, oneAnnBinder') = case oneAnnBinder of
+        Bind v    -> (varName v, oneAnnBinder)
+        Ignore ty -> let genVar = mkName $ mkNoShadowingStr "gen" in
+            (genVar, Bind (genVar:>ty))
+  let tyCon = UConDef interfaceName $ toNest [oneAnnBinder']
+  let funDefs = mkFunDefs (pos, varTypeName, interfaceName) record
+  return $ (UData tyCon [UConDef consName (toNest [Ignore record])]) : funDefs
+  where
+    mkFunDefs :: (SrcPos, Name, Name) -> UExpr -> [UDecl]
```

Comment:
Instead of defining `mkFunDefs` to take in a `UExpr` that must be a `URecordTy`, why not pass it the `LabeledItems` of types directly? In other words
```
mkFunDefs :: (SrcPos, Name, Name) -> LabeledItems UExpr -> [UDecl]
```

Reply:
Agreed, that would be a good simplification